plugin/trino-kafka-event-listener/src/main/java/io/trino/plugin/eventlistener/kafka/KafkaEventPublisher.java

```diff
@@ -56,22 +68,86 @@ public KafkaEventPublisher(KafkaEventListenerConfig config, KafkaProducerFactory
         String createdTopic = config.getCreatedTopicName().orElse("");
         String completedTopic = config.getCompletedTopicName().orElse("");
         String splitCompletedTopic = config.getSplitCompletedTopicName().orElse("");
+        kafkaProducerSupplier = () -> {
+            Map<String, String> configOverrides = config.getKafkaClientOverrides();
+            LOG.info("Creating Kafka publisher (SSL=%s) for topics: %s/%s with excluded fields: %s and kafka config overrides: %s",
+                    producerFactory instanceof SSLKafkaProducerFactory, createdTopic, completedTopic, config.getExcludedFields(), configOverrides);
+            KafkaProducer<String, String> kafkaProducer = producerFactory.producer(configOverrides);
+            try {
+                checkConnectivityToBrokers(config.getPublishCreatedEvent() ? createdTopic : completedTopic, config.getRequestTimeout().toMillis());
+            }
+            catch (Throwable e) {
+                try {
+                    kafkaProducer.close();
+                }
+                catch (Throwable closeException) {
+                    e.addSuppressed(closeException);
+                }
+                LOG.error(e, "Failed to initialize Kafka publisher.");
+                stats.kafkaPublisherFailedToInitialize();
+                throw new RuntimeException("Cannot connect to kafka broker", e);
+            }
+            return kafkaProducer;
+        };
+
+        try {
+            kafkaProducer = kafkaProducerSupplier.get();
+        }
+        catch (Exception e) {
+            if (config.getTerminateOnInitializationFailure()) {
+                throw e;
+            }
+        }
+
+        // schedule reconnecting
+        if (kafkaProducer == null) {
+            connectExecutor.schedule(new Runnable()
+            {
+                @Override
+                public void run()
+                {
+                    while (true) {
+                        try {
+                            // success
+                            KafkaProducer<String, String> kafkaProducer = kafkaProducerSupplier.get();
```

Comment:
Why do we need to recreate it? Will `kafkaProducer` stop working permanently in case of some intermittent issues?